core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/CloneWorkspaceContainerRunnerSpec.scala

```diff
@@ -0,0 +1,219 @@
+      specifiedTime shouldBe expectedTime
+      Future.successful(None)
+    }.when(runner)
+      .cloneSuccess(ArgumentMatchers.eq(workspaceId), ArgumentMatchers.eq(expectedTime))(
+        ArgumentMatchers.any[ExecutionContext]()
+      )
+
+    whenReady(runner.handleCloneResult(workspaceId, report))(
+      _ shouldBe WorkspaceManagerResourceMonitorRecord.Complete
+    )
+
+    verify(runner)
+      .cloneSuccess(ArgumentMatchers.eq(workspaceId), ArgumentMatchers.eq(expectedTime))(
+        ArgumentMatchers.any[ExecutionContext]()
+      )
+  }
+
+  it should "return a completed status if no user email is set on the job" in {
+    val runner = spy(
+      new CloneWorkspaceContainerRunner(
+        mock[SamDAO],
+        mock[WorkspaceManagerDAO],
+        mock[SlickDataSource],
+        mock[GoogleServicesDAO]
+      )
+    )
+
+    doAnswer(answer =>
+      Future.successful(Some(workspace.copy(errorMessage = Some(answer.getArgument(1).asInstanceOf[String]))))
+    ).when(runner)
+      .cloneFail(ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any[ExecutionContext]())
+
+    whenReady(runner(monitorRecord.copy(userEmail = None)))(
+      _ shouldBe WorkspaceManagerResourceMonitorRecord.Complete
+    )
+
+  }
+
+  it should "return Incomplete when the if the user context cannot be created" in {
+    val runner = spy(
+      new CloneWorkspaceContainerRunner(
+        mock[SamDAO],
+        mock[WorkspaceManagerDAO],
+        mock[SlickDataSource],
+        mock[GoogleServicesDAO]
+      )
+    )
+    doReturn(Future.failed(new org.broadinstitute.dsde.workbench.client.sam.ApiException()))
+      .when(runner)
+      .getUserCtx(ArgumentMatchers.eq(userEmail))(ArgumentMatchers.any())
+    doAnswer { answer =>
+      val errorMessage = answer.getArgument(1).asInstanceOf[String]
+      errorMessage should include(workspaceId.toString)
+      errorMessage should include(userEmail)
+      Future.successful(Some(workspace.copy(errorMessage = Some(errorMessage))))
+    }.when(runner)
+      .cloneFail(ArgumentMatchers.eq(workspaceId), ArgumentMatchers.any())(ArgumentMatchers.any[ExecutionContext]())
+
+    whenReady(runner(monitorRecord))(_ shouldBe WorkspaceManagerResourceMonitorRecord.Incomplete)
+    verify(runner).cloneFail(ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any[ExecutionContext]())
+  }
+
+  behavior of "handling the clone container report"
+
+  it should "set completedCloneWorkspaceFileTransfer on the workspace to the complete time in the report" in {
+    val runner = spy(
+      new CloneWorkspaceContainerRunner(
+        mock[SamDAO],
+        mock[WorkspaceManagerDAO],
+        mock[SlickDataSource],
+        mock[GoogleServicesDAO]
+      )
+    )
+    val completedTime = "2023-01-16T10:08:48.541-05:00"
+    val expectedTime = DateTime.parse(completedTime)
+    val report = new JobReport().status(JobReport.StatusEnum.SUCCEEDED).completed(completedTime)
+    doAnswer { answer =>
```

Comment:
It feels strange to me to mock out functions in the runner when that's the class we're testing. Do we need separate tests for `cloneFail` and `cloneSuccess`?

Reply:
The main reason cloneFail and cloneSuccess are there in the first place, is to allow method spying, since we're using slick directly (which is both a pain to mock, and tends to end up with very broad argument matchers when you do), instead of a proper dao/repository that can be mocked for workspaces (like we do for billing). 
I considered the idea of creating something along the lines of the 'BillingRepository', until I saw the `WorkspaceSupport` trait used by `WorkspaceService` and thought better of it for now.

Personally, I don't see much value in writing tests for those, since they are just a strict pass-through for updating the workspace with the given parameters, using straightforward already-in-place operations in slick.
LMK if you want to jump on a call or something to go over it further.

Reply:
That makes sense. Mocking slick is definitely a pain